### ISSUE TITLE
DistributedMinerWrapper has different behavior with DistributedLossWrapper

### ISSUE THREAD

**Issue by HaominJiang** [KevinMusgrave/pytorch-metric-learning#542]
Hi, I found that the distributed miner wrapper behaves differently from the distributed loss wrapper when world size is 1.
Code for reproduction:
```
miner = miners.PairMarginMiner(pos_margin=args.miner_pos_margin, neg_margin=args.miner_neg_margin)
loss_fn = losses.ContrastiveLoss(pos_margin=args.pos_margin, neg_margin=args.neg_margin)
loss_fn = losses.CrossBatchMemory(loss_fn, embedding_size=256, memory_size=args.memory_size, miner=miner)
miner = pml_dist.DistributedMinerWrapper(miner)
loss_fn = pml_dist.DistributedLossWrapper(loss=loss_fn)

hard_pairs = miner(embeddings, labels)
loss = loss_fn(embeddings, labels, hard_pairs)
```
It threw:
```
File "train.py", line 208, in train_one_epoch
    hard_pairs = miner(embeddings, labels)
File "/home/envs/torch1100/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
File "/home/envs/torch1100/lib/python3.8/site-packages/pytorch_metric_learning/utils/distributed.py", line 98, in forward
    all_emb, all_labels, labels, device = gather(embeddings, labels)
File "/home/envs/torch1100/lib/python3.8/site-packages/pytorch_metric_learning/utils/distributed.py", line 42, in gather
    all_emb = torch.cat([embeddings, dist_ref_emb], dim=0)
TypeError: expected Tensor as element 1 in argument 0, but got NoneType
```
I think this is because distributed loss wrapper has the statement `if world_size <= 1: return xxx`. When the world size is less than or equal to 1, `gather` is not performed. See  https://github.com/KevinMusgrave/pytorch-metric-learning/blob/64f47bafd12a0f68a7cffa4c70146918de92eabf/src/pytorch_metric_learning/utils/distributed.py#L94
However, the distributed miner wrapper does not have such a judgment statement. This will lead the distributed miner wrapper throws out exception while the distributed loss wraper works fine under `world_size = 1`.

**Comment by KevinMusgrave** [comment]
Fixed in version 1.6.3:

```
pip install pytorch-metric-learning==1.6.3
```

**Comment by HaominJiang** [comment]
Thanks for the prompt update！
